src/gentropy/dataset/study_locus.py

```diff
@@ -391,6 +393,53 @@ def _qc_subsignificant_associations(
             StudyLocusQualityCheck.SUBSIGNIFICANT_FLAG,
         )
 
+    def qc_abnormal_pips(
+        self: StudyLocus,
+        sum_pips_lower_threshold: float,
+        sum_pips_upper_threshold: float
+    ) -> StudyLocus:
+        """Filter study-locus by sum of posterior inclusion probabilities to ensure that the sum of PIPs is within a given range.
+
+        Args:
+            sum_pips_lower_threshold (float): Lower threshold for the sum of PIPs.
+            sum_pips_upper_threshold (float): Upper threshold for the sum of PIPs.
+
+        Returns:
+            StudyLocus: Filtered study-locus dataset.
+        """
+        # QC column might not be present so we have to be ready to handle it:
+        qc_select_expression = (
+            f.col("qualityControls")
+            if "qualityControls" in self.df.columns
+            else f.lit(None).cast(ArrayType(StringType()))
+        )
+
+        flag = (self.df.withColumn(
+            "sumPosteriorProbability",
+            f.aggregate(
+                f.col("locus"),
+                f.lit(0.0),
+                lambda acc, x: acc + x["posteriorProbability"]
+            )).filter((f.col("sumPosteriorProbability") < 0.99) | (f.col("sumPosteriorProbability") > 1.0))
```

Comment:
Instead of applying a filter + a join, you can calculate the sum, and based on that create a flag `pipOutOfRange` or something like this to update the quality flag array

Reply:
Makes sense, yeah - I'll implement this a bit later today